### ISSUE TITLE
feat: Improve the regexp, reference some standard projects

### ISSUE THREAD

**Issue by gxthrj** [apache/apisix-dashboard#1639]
Now, we use `jsonschema` to verfiy the fields of objects.

As @juzhiyuan suggested, Need to modify regexp refer to some standard projects . 

For example, verify fields such as name, password, email. This can avoid some mistakes.

**Comment by gxthrj** [comment]
Hi, guys.

Are there any good reference projects? Please list them here. I have searched many projects on github.
But there are no standard, only different needs. 

There are some regular libraries, but the regular expressions still need to be filled in by yourself.

**Comment by juzhiyuan** [comment]
Hello, please have this issue tagged :)

**Comment by juzhiyuan** [comment]
ya, here have some examples: https://projects.lukehaas.me/regexhub/

and @LiteSun, could you help to provide some rules from and?

**Comment by gxthrj** [comment]
Great~ but I think  https://projects.lukehaas.me/regexhub/ is not enough.

Maybe regex does have some standards or best practices, but meeting our needs is the most important thing.
We can only refer to the examples you give.

For example, we need regex-expr for ` ipv4` and `domain`, but it is only `ipv4`.
```
// now what we use
(^([1-9]?\d|1\d{2}|2[0-4]\d|25[0-5])(\.(25[0-5]|1\d{2}|2[0-4]\d|[1-9]?\d)){3}$|^(?![0-9.]+$)([a-zA-Z0-9_-]+)(\.[a-zA-Z0-9_-]+){0,}$)
```
```
// which is in lukehaas.me
/^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/
```

Another example about usename. We also need to support Chinese.
```
// now what we use is better
^[a-zA-Z\u4e00-\u9fa5][a-zA-Z0-9\u4e00-\u9fa5_-]{1,100}$
```

**Comment by LiteSun** [comment]
> ya, here have some examples: https://projects.lukehaas.me/regexhub/
> 
> and @LiteSun, could you help to provide some rules from antd?

ANTD uses the [async-validator](https://github.com/yiminghe/async-validator) library for field checking, if you want to see the regular rules you need to look through the source code, which is not very convenient

**Comment by bzp2010** [comment]
According to the project principles, the UI will directly use the Admin API, so this will no longer be an issue once the next renovation is complete. ref #2981 

If you have any questions, feel free to start a new issue!
